src_code/src/controllers/services/Authentication/SignIn.js

```diff
@@ -0,0 +1,82 @@
+ *  @returns {object} A response object showing whether request is success or not.
+ */
+async function SignIn(response, requestBody) {
+    try {
+        /** Validate sign in body to see if information they request to our endpoint is valid */
+        const errors = await SignInValidator(response, requestBody);
+        if(errors) {
+            return errors;
+        }
+
+        /** If validation pass, we need to destructure variable emailAddress from the request body for use. */
+        const {emailAddress} = requestBody;
+
+        /** Get user information based on emailAddress */
+        const userInfo = await dbHelper.GetUserInfoBasedOnEmailAddress(db, emailAddress.toLowerCase().trim());
+
+        /** If userInfo is a string, which means it is an error message */
+        if(typeof userInfo === "string") {
+            return responseBuilder.ServerError(response, userInfo);
+        }
+
+        /** If no error, return OK response */
+        return responseBuilder.BuildResponse(response, 200, {
+            message:"Sign in successfully.",
+            responseObject: userInfo
+        });
+    } catch(error) {
+        /** Loggin error, easy to debug */
+        console.log("ERROR: There is an error while siging in: ", error);
+        /** Return error message to client */
+        return responseBuilder.ServerError(response, "There is an error while signing in.");
+    }
+}
+
+async function SignInValidator(response, requestBody) {
+    try{
+        /** Destructure variables from the request body */
+        const {emailAddress, password} = requestBody;
+        
+        /** Check if user is filled in required fields */
+        if(!emailAddress || !password) {
+            return responseBuilder.MissingContent(response);
+        }
+
+        /** Check if email address is exists (or already sign up) */
+        const existUser = await db.select("ACCOUNT_PASSWORD").from("user_info").where("EMAIL_ADDRESS", "=", emailAddress.toLowerCase().trim()).first();
+        if(!existUser) {
+            return responseBuilder.BuildResponse(response, 404, {
+                message:"Invalid credentials. (username)"
+            })
+        }
+
+        /** If user is found with the email address, we need to check for their password if it correct */
+        if(!bcrypt.compareSync(password, existUser.ACCOUNT_PASSWORD)) {
+            return responseBuilder.BuildResponse(response, 404, {
+                message:"Invalid credentials. (password)"
+            })
+        }
+        
+        /** If all checks pass, return null to indicate validation success */
+        return null;
+    } catch(error){
+        /** Logging unexpected error, help for debug */
+        console.log("ERROR: There is an error while validating sign in: ", error);
+        /** Return error message to the client */
+        return responseBuilder.ServerError(response, "There is an error while validating sign in.");
+    }
+}
+
+module.exports = {
+    SignIn,
+}
```

Comment:
Let's add the newline here at the end of the file 👍Good explanations and steps through!